packages/fxa-settings/src/models/Account.ts

```diff
@@ -409,6 +421,22 @@ export class Account implements AccountData {
     return account.securityEvents;
   }
 
+  async getProfileInfo() {
+    try {
+      const { data } = await this.apolloClient.query({
+        query: GET_PROFILE_INFO,
+      });
+      const { account } = data;
+      return account as ProfileInfo;
+    } catch (err) {
+      const errno = (err as ApolloError).graphQLErrors[0].extensions?.errno;
```

Comment:
Q: Could this throw a non ApolloError?
Like, hypothetically if `apolloClient.query()` returns null, I'm not sure how the destructuring to `{ data }` or secondary destructuring to `{ account }` would behave and if it would throw a generic Error instance.
Then that tries to get cast to an ApolloError instance which possibly breaks if it tries accessing `graphQLErrors[]` array.
But I have no idea if any of that is possible/likely.

Possibly a non-issue since I only see `apolloClient.query` used in a few places (all in this file) but don't see a lot of validation on the responses).
```sh
git grep -En -B1 -A3 "this\.apolloClient.query" packages | cat
```

Reply:
And just because I'm now looking at grep logs, possibly interesting is that this is the only ApolloClient query that doesn't specify a `fetchPolicy`:

```sh
git grep -En -B1 -A3 "this\.apolloClient\.query\(" packages | cat | grep "fetchPolicy:"

packages/fxa-settings/src/models/Account.ts-402-        fetchPolicy: 'network-only',
packages/fxa-settings/src/models/Account.ts-417-      fetchPolicy: 'network-only',
packages/fxa-settings/src/models/Account.ts-451-        fetchPolicy: 'network-only',
packages/fxa-settings/src/models/Account.ts-828-      fetchPolicy: 'no-cache',
```

Reply:
All very good questions. This is my first foray into modifying the Account model. I modelled this new method on the one just below (`getRecoveryKeyBundle`) which has the same structure, and would 100% appreciate a good check on this (and my thoughts below).

If there are problems with destructuring to { data } or {account}, or if the type of account doesn't match with `ProfileInfo`, the try block should throw an error.

And I _think_ if `err` in the catch block is not of type `ApolloError` (or if there are other errors in defining `errno`), the last line of the catch block will be executed to handle the unknown error. 
```throw AuthUiErrors.UNEXPECTED_ERROR```

We could improve on the returned error, but I'm not sure how detailed we want to get here.